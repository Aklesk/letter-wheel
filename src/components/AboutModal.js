import React from 'react';
import PropTypes from 'prop-types';
import { Linking, View, Text } from 'react-native';
import Modal from 'react-native-modalbox';
import Hyperlink from 'react-native-hyperlink';

import AboutModalStyle from '../styles/AboutModalStyle';

const AboutModal = ({ isOpen, onClosed }) => (
  <Modal
    backButtonClose
    entry="bottom"
    isOpen={isOpen}
    onClosed={onClosed}
    position="center"
    style={AboutModalStyle.modal}
    swipeToClose
  >
    <View style={AboutModalStyle.container}>
      <Text style={AboutModalStyle.text}>
        {'Target Words is based on the popular newspaper game Target, and inspired by the ' +
        'long-abandoned Target app for Android by Ben Buxton.'}
      </Text>
      <Text style={AboutModalStyle.text}>
        {' '}
      </Text>
      <Hyperlink
        linkStyle={{ color: '#2980b9' }}
        onPress={() => {
          Linking.openURL('https://www.github.com/Fendrian/target-words/');
        }}
      >
        <Text style={AboutModalStyle.text}>
          {'Original copyright Â© 2017-2018 by Wolf Hatch. The source code for this app is released ' +
          'under the MIT license at https://github.com/Fendrian/target-words'}
        </Text>
      </Hyperlink>
      <Text style={AboutModalStyle.text}>
        {' '}
      </Text>
      <Text style={AboutModalStyle.text}>
        {'This work would not be remotely possible without the massive wealth of open-source ' +
        'and public domain resources that it is built upon, a full list of which is available ' +
        'with this project\'s source code on GitHub.'}
      </Text>
    </View>
  </Modal>
);

AboutModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClosed: PropTypes.func.isRequired,
};

export default AboutModal;
