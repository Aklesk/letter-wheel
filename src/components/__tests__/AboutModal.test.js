import React from 'react';
import { mount } from 'enzyme';
import { Linking } from 'react-native';
import Hyperlink from 'react-native-hyperlink';

import AboutModal from '../AboutModal';

jest.spyOn(Linking, 'openURL');

describe('About Modal', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    const render = mount((
      <AboutModal
        isOpen={false}
        onClosed={jest.fn()}
      />
    ));
    expect(render).toMatchSnapshot();
    render.setProps({ isOpen: true });
    expect(render).toMatchSnapshot();
  });

  it('calls onClosed when closed once all animation has finished', () => {
    const render = mount((
      <AboutModal
        isOpen
        onClosed={jest.fn()}
      />
    ));
    render.setProps({ isOpen: false });
    expect(render.props().onClosed).toHaveBeenCalledTimes(0);
    jest.advanceTimersByTime(1000);
    expect(render.props().onClosed).toHaveBeenCalledTimes(1);
    render.setProps({ isOpen: true });
    expect(render.props().onClosed).toHaveBeenCalledTimes(1);
  });

  it('links to the Target game GitHub repository', () => {
    const render = mount((
      <AboutModal
        isOpen
        onClosed={jest.fn()}
      />
    ));
    const link = render.find(Hyperlink).at(0);
    expect(link.props().onPress).toEqual(expect.any(Function));
    link.props().onPress();
    expect(Linking.openURL).toHaveBeenCalledTimes(1);
    expect(Linking.openURL).toHaveBeenCalledWith('https://github.com/buxtronix/target');
  });

  it('links to the GitHub repository', () => {
    const render = mount((
      <AboutModal
        isOpen
        onClosed={jest.fn()}
      />
    ));
    const link = render.find(Hyperlink).at(1);
    expect(link.props().onPress).toEqual(expect.any(Function));
    link.props().onPress();
    expect(Linking.openURL).toHaveBeenCalledTimes(1);
    expect(Linking.openURL).toHaveBeenCalledWith('https://www.github.com/Fendrian/letter-wheel/');
  });
});
