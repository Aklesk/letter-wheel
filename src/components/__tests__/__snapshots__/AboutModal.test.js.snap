// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Modal matches the snapshot 1`] = `
<AboutModal
  isOpen={false}
  onClosed={[MockFunction]}
>
  <ModalBox
    animationDuration={400}
    backButtonClose={true}
    backdrop={true}
    backdropColor="black"
    backdropContent={null}
    backdropOpacity={0.5}
    backdropPressToClose={true}
    coverScreen={false}
    easing={[Function]}
    entry="bottom"
    isOpen={false}
    keyboardTopOffset={22}
    onClosed={[MockFunction]}
    position="center"
    startOpen={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "width": 263.52941176470586,
      }
    }
    swipeThreshold={50}
    swipeToClose={true}
  >
    <View>
      <react-native-mock />
    </View>
  </ModalBox>
</AboutModal>
`;

exports[`About Modal matches the snapshot 2`] = `
<AboutModal
  isOpen={true}
  onClosed={[MockFunction]}
>
  <ModalBox
    animationDuration={400}
    backButtonClose={true}
    backdrop={true}
    backdropColor="black"
    backdropContent={null}
    backdropOpacity={0.5}
    backdropPressToClose={true}
    coverScreen={false}
    easing={[Function]}
    entry="bottom"
    isOpen={true}
    keyboardTopOffset={22}
    onClosed={[MockFunction]}
    position="center"
    startOpen={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "width": 263.52941176470586,
      }
    }
    swipeThreshold={50}
    swipeToClose={true}
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0)",
            "zIndex": 2,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <react-native-mock>
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <react-native-mock>
            <TouchableWithoutFeedback
              onPress={[Function]}
            >
              <react-native-mock>
                <AnimatedComponent
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "opacity": AnimatedValue {
                          "_animation": null,
                          "_children": Array [],
                          "_listeners": Object {},
                          "_offset": 0,
                          "_value": 0,
                        },
                      },
                    ]
                  }
                >
                  <View>
                    <react-native-mock>
                      <View
                        style={
                          Array [
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            Object {
                              "backgroundColor": "black",
                              "opacity": 0.5,
                            },
                          ]
                        }
                      >
                        <react-native-mock />
                      </View>
                    </react-native-mock>
                  </View>
                </AnimatedComponent>
              </react-native-mock>
            </TouchableWithoutFeedback>
            <AnimatedComponent
              onLayout={[Function]}
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                  },
                  Object {
                    "height": 768,
                    "width": 320,
                  },
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "width": 263.52941176470586,
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "translateY": AnimatedValue {
                          "_animation": null,
                          "_children": Array [],
                          "_listeners": Object {},
                          "_offset": 0,
                          "_value": 768,
                        },
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  },
                ]
              }
            >
              <View>
                <react-native-mock>
                  <View
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "borderRadius": 4.517647058823529,
                        "padding": 22.588235294117645,
                      }
                    }
                  >
                    <react-native-mock>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                          Target Words is based on the popular newspaper game Target, and inspired by the long-abandoned Target app for Android by Ben Buxton.
                        </react-native-mock>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                           
                        </react-native-mock>
                      </Text>
                      <_class
                        linkStyle={
                          Object {
                            "color": "#2980b9",
                          }
                        }
                        onPress={[Function]}
                      >
                        <Hyperlink
                          linkStyle={
                            Object {
                              "color": "#2980b9",
                            }
                          }
                          onPress={[Function]}
                        >
                          <View>
                            <react-native-mock>
                              <Text
                                key=".0"
                                onLongPress={[Function]}
                                style={
                                  Object {
                                    "color": "#000",
                                    "fontSize": 20.32941176470588,
                                  }
                                }
                              >
                                <react-native-mock>
                                  Original copyright Â© 2017-2018 by Wolf Hatch. The source code for this app is released under the MIT license at 
                                  <Text
                                    key="https://github.com/Fendrian/target-words112"
                                    onLongPress={[Function]}
                                    onPress={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#000",
                                          "fontSize": 20.32941176470588,
                                        },
                                        Object {
                                          "color": "#2980b9",
                                        },
                                      ]
                                    }
                                  >
                                    <react-native-mock>
                                      https://github.com/Fendrian/target-words
                                    </react-native-mock>
                                  </Text>
                                </react-native-mock>
                              </Text>
                            </react-native-mock>
                          </View>
                        </Hyperlink>
                      </_class>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                           
                        </react-native-mock>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#000",
                            "fontSize": 20.32941176470588,
                          }
                        }
                      >
                        <react-native-mock>
                          This work would not be remotely possible without the massive wealth of open-source and public domain resources that it is built upon, a full list of which is available with this project's source code on GitHub.
                        </react-native-mock>
                      </Text>
                    </react-native-mock>
                  </View>
                </react-native-mock>
              </View>
            </AnimatedComponent>
          </react-native-mock>
        </View>
      </react-native-mock>
    </View>
  </ModalBox>
</AboutModal>
`;
